### PR TITLE
RavenDB-20970 Move Tombstones to React

### PR DESCRIPTION
**Pull request by mateuszbartosik**
### Issue link
https://issues.hibernatingrhinos.com/issue/RavenDB-20970/Move-Settings-Tombstones-to-React

### Additional description
Moved Tombstones to React; visual improvements

### Type of change
- Optimization

### How risky is the change?
- Low

### Backward compatibility
- Non breaking change

### Is it platform specific issue?
- No

### Documentation update
- No documentation update is needed 

### Testing by Contributor
- Tests have been added that prove the fix is effective or that the feature works
- It has been verified by manual testing

### Testing by RavenDB QA team
- No special testing by RavenDB QA team is needed

### Is there any existing behavior change of other features due to this change?
- No

### UI work
- No UI work is needed

### CODE REVIEW

File: src/Raven.Studio/typescript/components/pages/database/settings/tombstones/TombstonesState.tsx [hunk 1]
```diff
@@ -0,0 +1,261 @@
...
+            <p>
+                By default, tombstones cleanup is scheduled to be carried out by the server every 5 minutes, unless
+                configured otherwise.
+            </p>
+            <p className="mb-0">
+                Upon clicking <strong>Force Cleanup</strong>, the action will be executed immediately - any tombstone
+                that can be removed will be deleted.
+            </p>
+        </Alert>
+    );
```
Comment (kalczur) [inline src/Raven.Studio/typescript/components/pages/database/settings/tombstones/TombstonesState.tsx]: It should be a separate component

File: src/Raven.Studio/typescript/components/hooks/useConfirm.tsx [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+        <Modal
+            isOpen={isOpen}
+            toggle={onCancel}
+            wrapClassName="bs5"
+            centered
+            contentClassName={`modal-border bulge-${actionColor ?? "secondary"}`}
+        >
+            <ModalBody className="vstack gap-4 position-relative">
+                <div className="text-center">
+                    <Icon icon={icon} color={actionColor} className="fs-1" margin="m-0" />
```
Comment (kalczur) [inline src/Raven.Studio/typescript/components/hooks/useConfirm.tsx:69]: `icon` and `actionColor` are optional. I think they should be required

File: src/Raven.Studio/typescript/components/hooks/useConfirm.tsx [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+                </div>
+                <div className="text-center lead">{title}</div>
+                {message ? message : null}
+            </ModalBody>
+            <ModalFooter>
+                <Button color="link" onClick={onCancel} className="text-muted">
+                    Cancel
+                </Button>
+                <Button color={actionColor} onClick={onConfirm} className="rounded-pill">
+                    {icon && <Icon icon={icon} />}
```
Comment (kalczur) [inline src/Raven.Studio/typescript/components/hooks/useConfirm.tsx]: If we have changed `icon` to required, then this checking is not needed

**Review by kalczur — CHANGES_REQUESTED** [review]

File: src/Raven.Studio/typescript/components/hooks/useConfirm.tsx (hunk 2 above)
Comment (kalczur) [reply src/Raven.Studio/typescript/components/hooks/useConfirm.tsx:69]: or set `actionColor` default to "secondary"

File: src/Raven.Studio/typescript/components/pages/database/settings/tombstones/TombstonesState.tsx (hunk 1 above)
Comment (mateuszbartosik) [reply src/Raven.Studio/typescript/components/pages/database/settings/tombstones/TombstonesState.tsx]: The alert has been moved to a separate component

File: src/Raven.Studio/typescript/components/hooks/useConfirm.tsx (hunk 2 above)
Comment (mateuszbartosik) [reply src/Raven.Studio/typescript/components/hooks/useConfirm.tsx:69]: Both `icon` and `actionColor` are required from now on

File: src/Raven.Studio/typescript/components/hooks/useConfirm.tsx (hunk 3 above)
Comment (mateuszbartosik) [reply src/Raven.Studio/typescript/components/hooks/useConfirm.tsx]: Done

**Review by kalczur — APPROVED** [review]

**Review by ml054 — APPROVED** [review]

**Pull request merged**